mp/src/game/server/momentum/mom_triggers.h

```diff
@@ -688,3 +688,52 @@ class CNoGrenadesZone : public CBaseTrigger
 
     bool m_bAirborneOnly;
 };
+
+// CTriggerMomentumCatapult
+class CTriggerMomentumCatapult : public CBaseMomentumTrigger
+{
+  public:
+    DECLARE_CLASS(CTriggerMomentumCatapult, CBaseMomentumTrigger);
+    DECLARE_DATADESC();
+
+    CTriggerMomentumCatapult();
+
+  public:
+    void OnStartTouch(CBaseEntity *) OVERRIDE;
+    void Spawn() OVERRIDE;
+    void Think() OVERRIDE;
+
+    int DrawDebugTextOverlays() OVERRIDE;
+
+  private:
+    Vector CalculateLaunchVelocity(CBaseEntity *);
+    Vector CalculateLaunchVelocityExact(CBaseEntity *);
+    void Launch(CBaseEntity *);
+    void LaunchAtDirection(CBaseEntity *);
+    void LaunchAtTarget(CBaseEntity *);
+
+  private:
+
+    enum
+    {
+        BEST=0,
+        SOLUTION_ONE,
+        SOLUTION_TWO,
+    };
```

Comment:
It's good to name your enums:
```suggestion
    enum ExactVelocityChoice_t
    {
        BEST = 0,
        SOLUTION_ONE,
        SOLUTION_TWO,
    };
```